File: feature/station23/src/main/java/com/example/instalearnenglish/feature/station23/Station23Activity.java
package com.example.instalearnenglish.feature.station23;

import android.os.Bundle;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager2.widget.ViewPager2;

import com.example.instalearnenglish.feature.home.tools.DictionaryDialogFragment;
import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;

public class Station23Activity extends AppCompatActivity {

    private ViewPager2 viewPager;
    private TabLayout tabLayout;
    private Station23PagerAdapter pagerAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_station23);

        viewPager = findViewById(R.id.viewPager);
        tabLayout = findViewById(R.id.tabLayout);
        ImageButton btnBack = findViewById(R.id.btn_back);
        ImageButton btnDictionary = findViewById(R.id.btn_dictionary);
        TextView toolbarTitle = findViewById(R.id.toolbar_title);

        // Get stationId from Intent using the correct key "LEVEL" (consistent with HomeActivity)
        int stationId = getIntent().getIntExtra("LEVEL", 2);

        if (toolbarTitle != null) {
            toolbarTitle.setText(stationId == 2 ? "Trạm 2: Sân bay" : "Trạm 3: Di chuyển");
        }

        if (btnBack != null) {
            btnBack.setOnClickListener(v -> onBackPressed());
        }

        if (btnDictionary != null) {
            btnDictionary.setOnClickListener(v -> {
                DictionaryDialogFragment dialogFragment = new DictionaryDialogFragment();
                dialogFragment.show(getSupportFragmentManager(), "DictionaryDialog");
            });
        }

        // Pass stationId to the PagerAdapter
        pagerAdapter = new Station23PagerAdapter(this, stationId);
        viewPager.setAdapter(pagerAdapter);
        
        // Vô hiệu hóa lướt ngang của ViewPager để ưu tiên lướt các thẻ Tip
        viewPager.setUserInputEnabled(false);

        new TabLayoutMediator(tabLayout, viewPager,
                (tab, position) -> {
                    switch (position) {
                        case 0: tab.setText("Travel Tip"); break;
                        case 1: tab.setText("Vocabulary"); break;
                        case 2: tab.setText("Simulator"); break;
                        case 3: tab.setText("Mini-Game"); break;
                    }
                }
        ).attach();
    }
}

File: feature/station23/src/main/java/com/example/instalearnenglish/feature/station23/Station23Activity.kt
package com.example.instalearnenglish.feature.station23

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.instalearnenglish.core.ui.adapter.LessonPagerAdapter
import com.example.instalearnenglish.feature.station23.databinding.FeatureStation23MainBinding
import com.google.android.material.tabs.TabLayoutMediator

class Station23Activity : AppCompatActivity() {

    private lateinit var binding: FeatureStation23MainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        try {
            binding = FeatureStation23MainBinding.inflate(layoutInflater)
            setContentView(binding.root)

            val level = intent.getIntExtra("LEVEL", 2)
            setupToolbar(level)
            setupViewPager()
        } catch (e: Exception) {
            Toast.makeText(this, "Lỗi khởi tạo Trạm 2/3: ${e.message}", Toast.LENGTH_SHORT).show()
            finish()
        }
    }

    private fun setupToolbar(level: Int) {
        setSupportActionBar(binding.toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        binding.toolbar.title = if (level == 2) "Trạm 2: Tại Sân bay" else "Trạm 3: Di chuyển"
        binding.toolbar.setNavigationOnClickListener { 
            val intent = Intent()
            intent.setClassName(this, "com.example.instalearnenglish.feature.home.HomeActivity")
            startActivity(intent)
            finish()
        }
    }

    private fun setupViewPager() {
        // Gán Adapter trước khi dùng Mediator
        binding.viewPager.adapter = LessonPagerAdapter(this)
        
        val tabTitles = listOf("Tip", "Vocab", "Simulation", "Game")
        TabLayoutMediator(binding.tabLayout, binding.viewPager) { tab, position ->
            tab.text = tabTitles[position]
        }.attach()
    }
}
